extras/build_hashes.py

```diff
@@ -11,5 +14,11 @@
             for filename in os.listdir(entry_path):
                 path = os.path.join(entry_path, filename)
                 hash = hashlib.sha256(open(path).read()).hexdigest()
-                print path, hash
-                output.write('{} {}\n'.format(hash, os.path.basename(path)))
+                if hash not in hashes:
+                    print path, hash
+                    output.write('{} {}\n'.format(hash, os.path.basename(path)))
+                    hashes[hash] = path
+                else:
+                    print "Error: {} and {} have the same hash ({})...Exiting.".format(path, hashes[hash], hash)
```

Comment:
This should not happen. This is a fallback where the URL we retrieve is not the actual script file but e.g. a 404 page, but the status code was not 404. It's just a safety net to point out that something fishy is going on. The case is resolved by examining both URLs and determining which one is wrong.